src/Microsoft.Diagnostics.EventFlow.Inputs.EventSource/EventSourceInput.cs

```diff
@@ -57,6 +58,7 @@ public EventSourceInput(IReadOnlyCollection<EventSourceConfiguration> eventSourc
         }
 
         public IEnumerable<EventSourceConfiguration> EventSources { get; private set; }
+        public Task Initialization { get; private set; }
```

Comment:
The main reason is somebody might really care about a particular "service initialization" event to be captured and this is not guaranteed until the initialization has been completed. But your question prompted me to seek a better way and indeed there is: we have an interface, `IRequiresActivation` that can be used for the pipeline to wait for initialization tasks to complete. I will change the implementation to use that.